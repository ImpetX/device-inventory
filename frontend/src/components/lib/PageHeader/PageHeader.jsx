import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import grey from '@material-ui/core/colors/grey';
import makeStyles from '@material-ui/styles/makeStyles';
import Button from '../Button';

const PageHeader = forwardRef((props, ref) => {
  const useStyles = makeStyles({
    PageHeader: {
      borderBottom: `.0625rem solid ${grey[400]}`,
      display: 'flex',
      flexWrap: 'nowrap',
      overflow: 'hidden',
      padding: '1.25rem .9375rem',
    },

    PageHeader__left: {
      flex: 1,
    },
  });

  const classes = useStyles();

  const {
    buttonLabel,
    buttonIcon,
    buttonVariant,
    headerText,
    onButtonClick,
  } = props;

  return (
    <div className={classes.PageHeader} ref={ref}>
      <div className={classes.PageHeader__left}>
        <Typography variant="h4" component="h2">
          {headerText}
        </Typography>
      </div>
      <div>
        <Button
          color="primary"
          icon={buttonIcon}
          label={buttonLabel}
          variant={buttonVariant}
          onClick={e => onButtonClick(e)}
        />
      </div>
    </div>
  );
});

PageHeader.propTypes = {
  headerText: PropTypes.string.isRequired,
  buttonLabel: PropTypes.string.isRequired,
  buttonIcon: PropTypes.element,
  buttonVariant: PropTypes.oneOf(['outlined', 'contained']),
  onButtonClick: PropTypes.func.isRequired,
};

PageHeader.defaultProps = {
  buttonIcon: null,
  buttonVariant: 'contained',
};

export default PageHeader;
