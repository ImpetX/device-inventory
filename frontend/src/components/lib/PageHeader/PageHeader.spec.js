import React from 'react';
import { shallow } from 'enzyme';
import PageHeader from '.';
import { Button } from '..';

describe('Page Header', () => {
  const mockFn = jest.fn();

  const component = (buttonIcon, buttonVariant) => {
    return shallow(
      <PageHeader
        headerText="Header"
        buttonLabel="button"
        buttonIcon={buttonIcon}
        buttonVariant={buttonVariant}
        onButtonClick={mockFn}
      />
    );
  };

  it('should render correctly', () => {
    expect(component()).toMatchSnapshot();
  });

  it('should call the click handler on click event', () => {
    component()
      .find(Button)
      .simulate('click');

    expect(mockFn).toHaveBeenCalled();
  });
});
