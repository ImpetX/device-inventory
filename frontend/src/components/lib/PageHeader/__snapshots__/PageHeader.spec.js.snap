// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Header should render correctly 1`] = `
<div
  className="makeStyles-PageHeader-1"
  id="PageHeader"
>
  <div
    className="makeStyles-PageHeader__left-2"
  >
    <WithStyles(ForwardRef(Typography))
      component="h2"
      variant="h4"
    >
      Header
    </WithStyles(ForwardRef(Typography))>
  </div>
  <div>
    <Button
      color="primary"
      icon={null}
      label="button"
      onClick={[Function]}
      variant="contained"
    />
  </div>
</div>
`;
